### PR TITLE
Fixing issue #552

### LINKED ISSUE #522: 🐛 [Bug] Segfault when wrong dimensions are supplied, after generating a node with empty output dimensions
##  Bug Description

trtorchc segfaults after generating empty output dimensions for adaptive pooling.
This happens because wrong number of dimensions are being supplied for input. 
Next op crashes: 
DEBUG: Original shape: [1, 1024], reshaping to: [1, 1024, 1, 1]
DEBUG: Weights: [1024]
    Number of input maps: 1024
    Number of output maps: 1024
    Element shape: [1]
DEBUG: Weights: [1024]
    Number of input maps: 1024
    Number of output maps: 1024
    Element shape: [1]
DEBUG: Weights: [1024]
    Number of input maps: 1024
    Number of output maps: 1024
    Element shape: [1]
DEBUG: Original shape: [1, 1024, 1, 1], reshaping to: [1, 1024]
INFO: [TRTorch Conversion Context] - Adding Layer %y.4 : Tensor = aten::adaptive_avg_pool1d(%x.13, %5) # /usr/local/lib/python3.6/dist-packages/torch/nn/modules/pooling.py:1098:0 (ctx.AddLayer)
DEBUG: [TRTorch Conversion Context] - Node input is an already converted tensor
DEBUG: [TRTorch Conversion Context] - Node input is a result of a previously evaluated value
DEBUG: Frozen tensor shape: [1, 1024]
DEBUG: Frozen tensor shape: [1, 1024]
ERROR: [TRTorch Conversion Context] - 3: %y.4 : Tensor = aten::adaptive_avg_pool1d(%x.13, %5) # /usr/local/lib/python3.6/dist-packages/torch/nn/modules/pooling.py:1098:0: no dimensions to be reduced
DEBUG: GlobalPoolingConverter: Output tensor shape: []
INFO: [TRTorch Conversion Context] - Adding Layer %input0.3 : Tensor = aten::transpose(%y.4, %12, %3) # /raid/git/NeMo/nemo/collections/asr/parts/submodules/jasper.py:423:0 (ctx.AddLayer)
DEBUG: [TRTorch Conversion Context] - Node input is an already converted tensor
DEBUG: [TRTorch Conversion Context] - Node input is a result of a previously evaluated value
DEBUG: [TRTorch Conversion Context] - Node input is a result of a previously evaluated value
ERROR: [TRTorch Conversion Context] - 3: %y.4 : Tensor = aten::adaptive_avg_pool1d(%x.13, %5) # /usr/local/lib/python3.6/dist-packages/torch/nn/modules/pooling.py:1098:0: no dimensions to be reduced
DEBUG: Frozen tensor shape: []
ERROR: [TRTorch Conversion Context] - 3: %y.4 : Tensor = aten::adaptive_avg_pool1d(%x.13, %5) # /usr/local/lib/python3.6/dist-packages/torch/nn/modules/pooling.py:1098:0: no dimensions to be reduced

## To Reproduce

1. Run trtorchc on this : https://drive.google.com/file/d/1SvnrYzPkBdS7OUgdNwPvx7V_u-wZAR05/view?usp=sharing

## Expected behavior

Should not segfault; error message may give a hint on dimensions or number of dimensions mismatch.

## Environment

> Build information about the TRTorch compiler can be found by turning on debug messages

 - TRTorch Version (e.g. 0.2.0): 
 - PyTorch Version (e.g. 1.0):
 - CPU Architecture:
 - OS (e.g., Linux):
 - How you installed PyTorch (`conda`, `pip`, `libtorch`, source):
 - Build command you used (if compiling from source):
 - Are you using local sources or building from archives:
 - Python version:
 - CUDA version:
 - GPU models and configuration:
 - Any other relevant information:

### LINKED ISSUE #552: ✨[Feature] Dynamic shape support in Automatic fallback
**Is your feature request related to a problem? Please describe.**
Support dynamic shapes in the input via automatic fallback.

**Describe the solution you'd like**
Currently the compilation fails when automatic fallback is enabled. Throw an error for this configuration or add support for dynamic shapes as a feature.

### PR DESCRIPTION
**Pull request by borisfom**
Signed-off-by: Boris Fomitchev <bfomitchev@nvidia.com>

# Description
Fixes #522  (issue)

## Type of change

Please delete options that are not relevant and/or add your own.

- Bug fix (non-breaking change which fixes an issue)
Added checks to bail out in case of obviously wrong dimension counts.

# Checklist:

- [ ] My code follows the style guidelines of this project (You can use the linters)
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas and hacks
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests to verify my fix or my feature
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: core/conversion/converters/impl/batch_norm.cpp [hunk 1]
```diff
@@ -40,6 +40,7 @@ auto batch_norm_registrations TRTORCH_UNUSED = RegisterNodeConversionPatterns().
       LOG_DEBUG("momentum disregarded");
       LOG_DEBUG("training disregarded");
       LOG_DEBUG("cudnn disregarded");
+      TRTORCH_CHECK(orig_shape.nbDims > 2 , "Unable to create batch normalization layer from node: " << *n);
```
Comment (peri044) [inline core/conversion/converters/impl/batch_norm.cpp:43]: In the current way, TRTORCH_CHECK just prints orig_shape.nbDims > 2 failed. Unable to create ... 
 Just to be more explicit, can we change these messages from "Unable to ..." to "Batch norm layer requires the input to be atleast 3 dimensional" ?
 Similar to other files as well.

**Review by peri044 — COMMENTED** [review]
LGTM. A minor message change
